src/realm_dart_sync.cpp

```diff
@@ -21,32 +21,29 @@
 #include "realm_dart.hpp"
 #include "realm_dart_sync.h"
 
-RLM_API void realm_dart_http_request_callback(realm_userdata_t userdata, const realm_http_request_t request, void* request_context) {
-    // the pointers in error are to stack values, we need to make copies and move them into the scheduler invocation
+RLM_API void realm_dart_http_request_callback(realm_userdata_t userdata, realm_http_request_t request, void* request_context) {
+    // the pointers in request are to stack values, we need to make copies and move them into the scheduler invocation
     struct request_copy_buf {
         std::string url;
         std::string body;
         std::map<std::string, std::string> headers;
```

Comment:
FYI, it is probably better to use a `std::deque<std::pair<std::string, std::string>>` here. You could also use a `std::vector<...>` as long as you `reserve()` the correct size up front, or do two passes, one to push everything, and another to push collect the addresses of the strings. You just need to avoid a growth event after collecting the addresses.